### PR TITLE
Fuel - new recipe

### PR DESCRIPTION
**Pull request by mazbrili**
new recipe..

### CODE REVIEW

File: dev-vcs/fuel/fuel-1.0.1.recipe [hunk 1]
```diff
@@ -0,0 +1,89 @@
+SUMMARY="A QT Fossil GUI"
+DESCRIPTION=" Fuel is cross-platform GUI front-end for the excellent Fossil SCM tool \
+written in Qt. Fuel runs on Windows, OSX as well as various flavors of Unix including \
+Linux and FreeBSD"
```
Comment (fbrosson) [inline dev-vcs/fuel/fuel-1.0.1.recipe]:
```diff
-DESCRIPTION=" Fuel is cross-platform GUI front-end for the excellent Fossil SCM tool \
-written in Qt. Fuel runs on Windows, OSX as well as various flavors of Unix including \
-Linux and FreeBSD"
+DESCRIPTION="Fuel is cross-platform GUI front-end for the excellent Fossil SCM \
+tool written in Qt. Fuel runs on Windows, OSX as well as various flavors of \
+Unix including Linux and FreeBSD."
```
Changes:
* No space before `Fuel`
* line splits to have lines <= 80 chars
* full-stop after `FreeBSD`

File: dev-vcs/fuel/fuel-1.0.1.recipe [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://fuel-scm.org/files/releases/fuel-$portVersion.tar.gz"
+CHECKSUM_SHA256="4e463cb9f3cf9c31c7ae95ce76b558502c0b07e1dd4109da714997f0dc9a65db"
+PATCHES="fuel-1.0.1.patchset"
+SOURCE_FILENAME="fuel-$portVersion.tar.gz"
+SOURCE_DIR="fuel-$portVersion"
+ADDITIONAL_FILES="fuel.rdef.in"
+
+ARCHITECTURES="x86 x86_64"
```
Comment (fbrosson) [inline dev-vcs/fuel/fuel-1.0.1.recipe]:
```diff
 CHECKSUM_SHA256="4e463cb9f3cf9c31c7ae95ce76b558502c0b07e1dd4109da714997f0dc9a65db"
-PATCHES="fuel-1.0.1.patchset"
 SOURCE_FILENAME="fuel-$portVersion.tar.gz"
-SOURCE_DIR="fuel-$portVersion"
+PATCHES="fuel-$portVersion.patchset"
 ADDITIONAL_FILES="fuel.rdef.in"
 
-ARCHITECTURES="x86 x86_64"
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
 if [ "$targetArchitecture" = x86_gcc2 ]; then
 SECONDARY_ARCHITECTURES="x86"
 fi
```
(`SOURCE_DIR` is not needed here because its default value is correct.)

File: dev-vcs/fuel/fuel-1.0.1.recipe (hunk 1 above)
Comment (mazbrili) [reply dev-vcs/fuel/fuel-1.0.1.recipe]: omg.. i forget again.. ok

File: dev-vcs/fuel/fuel-1.0.1.recipe (hunk 2 above)
Comment (mazbrili) [reply dev-vcs/fuel/fuel-1.0.1.recipe]: ..sorry..  ok changed

File: dev-vcs/fuel/fuel-1.0.1.recipe [hunk 3]
```diff
@@ -0,0 +1,88 @@
+SUMMARY="A QT Fossil GUI"
```
Comment (diversys) [inline dev-vcs/fuel/fuel-1.0.1.recipe]: -A QT Fossil GUI
+A Qt Fossil GUI

Comment (mazbrili) [reply dev-vcs/fuel/fuel-1.0.1.recipe]: ok done

**Comment by extrowerk** [comment]
Please use descriptive commit messages.

**Comment by mazbrili** [comment]
> Please use descriptive commit messages.

ok.. i will do it nextime

File: dev-vcs/fuel/fuel-1.0.1.recipe [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+tool written in Qt. Fuel runs on Windows, OSX as well as various flavors of \
+Unix including Linux and FreeBSD."
+HOMEPAGE="https://fuel-scm.org/fossil/home/"
+COPYRIGHT="2018 Kostas Karanikolas"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://fuel-scm.org/files/releases/fuel-$portVersion.tar.gz"
+CHECKSUM_SHA256="4e463cb9f3cf9c31c7ae95ce76b558502c0b07e1dd4109da714997f0dc9a65db"
+PATCHES="fuel-$portVersion.patchset"
+SOURCE_FILENAME="fuel-$portVersion.tar.gz"
```
Comment (fbrosson) [inline dev-vcs/fuel/fuel-1.0.1.recipe]:
```diff
 CHECKSUM_SHA256="4e463cb9f3cf9c31c7ae95ce76b558502c0b07e1dd4109da714997f0dc9a65db"
-PATCHES="fuel-$portVersion.patchset"
 SOURCE_FILENAME="fuel-$portVersion.tar.gz"
+PATCHES="fuel-$portVersion.patchset"
 ADDITIONAL_FILES="fuel.rdef.in"
```
[HaikuPorter-Guidelines#ordering](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering)
Thanks :)

Comment (mazbrili) [reply dev-vcs/fuel/fuel-1.0.1.recipe]: ok done.. thanks

Comment (fbrosson) [reply dev-vcs/fuel/fuel-1.0.1.recipe]: I was wondering why I had attached my comment to the line which had `SOURCE_FILENAME`...
I just found the answer: I had noticed that we could remove that line instead of moving `PATCHES`.. but then I must have forgotten we could remove `SOURCE_FILENAME`, so I ended editing the comment the way I had planed to do it before noticing `SOURCE_FILENAME` could be removed.
```diff
-SOURCE_FILENAME="fuel-$portVersion.tar.gz"
```
I'm sorry about that. Could you remove the line with the `SOURCE_FILENAME` ?
Thanks again :)

BTW, the reason why we can remove this is because this is the original name of the archive, obviously.
And we prefer to only use a `SOURCE_FILENAME` if it's required.

Comment (mazbrili) [reply dev-vcs/fuel/fuel-1.0.1.recipe]: ok.. done mate

File: dev-vcs/fuel/fuel-1.0.1.recipe [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:qmake$secondaryArchSuffix >= 5
+	"
+
+defineDebugInfoPackage fuel$secondaryArchSuffix \
+	"$appsDir"/Fuel
```
[the remark this answers is not in the document]
Comment (mazbrili) [reply dev-vcs/fuel/fuel-1.0.1.recipe]: ok debuginfo have been remove. thanks..
i'm feel stupid. thereis so many think i'm not yet know about build recipe.. still learning

**Comment by fbrosson** [comment]
The app builds fine on x86_64 and x86, but for some unknown reason I can't start it from the menu. I need to start it from the command line.
And I also get an error: `Could not start Fossil executable: 'fossil'`
And if I open the preferences dialog and enter `/bin/fossil` then I still get a similar error message: `Could not start Fossil executable: '/bin/fossil'`

I also tried starting inside a directory where I had just created a fossil repo with `fossil init foobar`, but this did not help.

Shall we keep the PR open for now, until someone finds a fix for these issues?
Or would you prefer to have it merged as is?
(Well, maybe we can wait one or two days and then merge even if we did not find a fix.)

**Comment by mazbrili** [comment]
> The app builds fine on x86_64 and x86, but for some unknown reason I can't start it from the menu. I need to start it from the command line.
> And I also get an error: `Could not start Fossil executable: 'fossil'`
> And if I open the preferences dialog and enter `/bin/fossil` then I still get a similar error message: `Could not start Fossil executable: '/bin/fossil'`
> 
> I also tried starting inside a directory where I had just created a fossil repo with `fossil init foobar`, but this did not help.
> 
> Shall we keep the PR open for now, until someone finds a fix for these issues?
> Or would you prefer to have it merged as is?
> (Well, maybe we can wait one or two days and then merge even if we did not find a fix.)
i found that issue too .. on some hrev.. i ask in haiku telegram but seem no one have interest to check...
do you have idea?..

File: dev-vcs/fuel/fuel-1.0.1.recipe [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+BUILD()
+{
+	qmake fuel.pro
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p "$appsDir"
```
Comment (diversys) [inline dev-vcs/fuel/fuel-1.0.1.recipe]: You don't need quotes here and below.

Comment (fbrosson) [reply dev-vcs/fuel/fuel-1.0.1.recipe]: We **_currently_** don't need them. But if someone chooses to use a `TREE_PATH` with a space character, and haikuporter's `--nochroot` switch is implemented in a way that allows to build recipes, then the quotes would be necessary. (Yeah, I know, this will probably never happen because a runtimes compiled outside of the chroot would need to be changed to make their paths correct, but this is IMHO not a good reason to not use the quotes here.)
To summarise:
1) We have lots of quotes in places where they will never be required.
2) In places like here, were it would be nice to start using them, I don't see why we would drop them.

I've given up suggesting this to others. I use the quotes in the recipes I edit. If this can serve as a model to other contributors, then I'm happy. We don't need to discourage the use of the quotes imo.

**Comment by pulkomandy** [comment]
Usually programs starting only from the command line are caused by expectations about the working directory and adding a chdir(dirname(argv[0])); at the start of the main() function will help. Look for other patches doing that if you need an example

**Comment by mazbrili** [comment]
> Sorry @mazbrili, I've no idea at the moment on how to fix the remaining issues.
> 
> * `Could not start Fossil executable: 'fossil'`
> * Fuel only starts when launched from the command line.

it is okay... thanks a lot for your assist..

**Comment by Begasus** [comment]
> Sorry @mazbrili, I've no idea at the moment on how to fix the remaining issues.
> 
>     * `Could not start Fossil executable: 'fossil'`
> 
>     * Fuel only starts when launched from the command line.

Works ok for me on secondary architecture, starts up from the Deskbar Menu fine

**Comment by fbrosson** [comment]
> Works ok for me on secondary architecture, starts up from the Deskbar Menu fine

That's cool! Thanks @Begasus!
I guess the issues I'm having on my x86_gcc2h system are not due to the recipe, which LGTM, but to my system.
So let's merge!

**Comment by mazbrili** [comment]
Many thanks too.. for @fbrosson  and @Begasus .. i'm happy contributing..
